include/libtorrent/packet_pool.hpp

```diff
@@ -79,16 +79,24 @@ namespace libtorrent
 	};
 
 
-	struct packet_deleter
+	struct packet_mem_manager
 	{
-		void operator()(packet* handle) const
+		// deleter for std::unique_ptr
+		void operator()(packet* p) const
 		{
-			handle->~packet();
-			std::free(handle);
+			p->~packet();
+			std::free(p);
+		}
+
+		static packet *create_packet(int size)
+		{
+			packet *p = static_cast<packet*>(std::malloc(sizeof(packet) + size));
+			new (p) packet();
+			return p;
```

Comment:
It's not clear to me what the benefit of conflating these two is. This static member function does not need access to any members in this class. it should probably just be a free function. I think calling this struct a deleter is very helpful too, because that's what the standard refers to by.